File: Java Barcode Scanner ME/app/src/main/java/de/dimedis/mobileentry/model/OnUserOfflineLoginCompleted.java
package de.dimedis.mobileentry.model;

import de.dimedis.lmlib.myinterfaces.Session;
import de.dimedis.mobileentry.bbapi.Constants;
import de.dimedis.mobileentry.util.SessionUtils;

public class OnUserOfflineLoginCompleted {
    private Session session;

    public OnUserOfflineLoginCompleted(Session session) {
        setSession(session);
    }

    public void setSession(Session session) {
        this.session = session;
    }

    public Session getSession() {
        return session;
    }

    public boolean isSuccessfulLogin() {
        return session.getStatus().equalsIgnoreCase(Constants.OFFLINE_LOGIN_SUCCESS_STATUS);
    }

    public void saveUserSessionPrefs() {
        SessionUtils.setUsersParam(getSession());
        SessionUtils.storeOfflineSession(getSession());
    }
}

File: Kotlin Migration/NTMobileEntry/app/src/main/java/de/dimedis/mobileentry/model/OnUserOfflineLoginCompleted.kt
package de.dimedis.mobileentry.model

import de.dimedis.lmlib.myinterfaces.Session
import de.dimedis.mobileentry.bbapi.Constants
import de.dimedis.mobileentry.util.SessionUtils

class OnUserOfflineLoginCompleted(session: Session?) {
    private var session: Session? = null

    init {
        setSession(session)
    }

    fun setSession(session: Session?) {
        this.session = session
    }

    fun getSession(): Session? {
        return session
    }

    fun isSuccessfulLogin(): Boolean {
        return session!!.getStatus().equals(Constants.OFFLINE_LOGIN_SUCCESS_STATUS, ignoreCase = true)
    }

    fun saveUserSessionPrefs() {
        getSession()?.let { SessionUtils.setUsersParam(it) }
        getSession()?.let { SessionUtils.storeOfflineSession(it) }
    }
}